lib/puppet/provider/service/windows.rb

```diff
@@ -92,25 +92,27 @@ def stop
   end
 
   def status
-    return :stopped unless Puppet::Util::Windows::Service.exists?(@resource[:name])
```

Comment:
I'd suggest keeping lines 95-113 the same as I think only the guard clause needs changing?

```diff
I'd suggest keeping lines 95-113 the same as I think only the guard clause needs changing?

```suggestion
-    return :stopped unless Puppet::Util::Windows::Service.exists?(@resource[:name])
+    return :absent unless Puppet::Util::Windows::Service.exists?(@resource[:name])
```

Reply:
That is much simpler, good catch.